compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3615,6 +3614,58 @@ public void visit(BLangInvocation iExpr) {
         fixTypeCastInTypeParamInvocation(iExpr, genIExpr);
     }
 
+    @Override
+    public void visit(BLangInvocation.BLangActionInvocation actionInvocation) {
```

Comment:
Aren't we duplicating most of the logic from `visit(BLangInvocation iExpr)` here? Can't we use a common method instead?

Reply:
Yeah, it's duplicated. Will see about moving it to a common place

Reply:
Addressed with ccf1fe7